src/libraries/System.Net.Http/src/System/Net/Http/Headers/CacheControlHeaderValue.cs

```diff
@@ -26,7 +26,6 @@ public class CacheControlHeaderValue : ICloneable
         private const string sharedMaxAgeString = "s-maxage";
 
         private static readonly HttpHeaderParser s_nameValueListParser = GenericHeaderParser.MultipleValueNameValueParser;
-        private static readonly Action<string> s_checkIsValidToken = CheckIsValidToken;
 
         private bool _noCache;
         private ObjectCollection<string>? _noCacheHeaders;
```

Comment:
Since we've changed the type we allocate for this from ObjectCollection to TokenObjectCollection, we should also change the field definition to use TokenObjectCollection. 

Similarly for other instances of TokenObjectCollection and UnvalidatedObjectCollection below.

Reply:
Going by the .NET 5 [codegen](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKBoGIA7AVwBsXtgWYACGBjrjRpluADRoBvGtxncADlACWAN2wYeAeWAArGGAwBhCGz0ZFEBgB5i5AAwA+APzcA+gDNFMFgBNyAbmlZBRU1HgBVBlUWRW9Q7y1dfSMTfXMrGwdnd08fUgDaalluYgBmYpRuAFlyAAoM7gQASm4AXntXDy9fRwA6ADVsaNj1Gqb8otLyqtI6uwbmto6c71JegaHQ0cbx2UCZYgqASQZFDBrGve4pQqLZbK7yVu4GGAB3bgiomLiE02SuVIWax2eznHa3JZdUhPF7vT6Db7qeI6P7GAFmIEZUHbS4AXxo+IKigY6ig/BY3FwMEGMG8xWh8I2SN+STRpjSlgAKu0QNwWYY2YCrNzuK8ABYwWDcTncXlkSSXSYQZSSpTeHgHbjrREwGoy04wfAXG4ya4QmSKNzcGoG/DcRS4Z6sFjG81XS5ujBiqAQd6w7gAQSgAHMmPg+BgAHLOgCiCDAMDkGIYNX44Ygbht6iNOJNt0JRUJhJoxNJ5O4HFwGCg2H09L5KNZKWTXPsCrzk0r1drGCm2uGuv12dzuKAA=), it does not seem to make any difference (M1 and M2 have same codegen) when nullable fields are involed. In non-nullable (unsafe) mode, it does [shaves off](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKBrIAIANGgbxvo/oAcoBLAN2wYY9APLAAVjDAYAwhAA2C6Rl4QAdgB5i5AAwA+egH0AZrxgKAJuQDc7TjwFCRAVXWCFvS88vipM+SUVNS0dA2MzC0tSO1pqTnpiAGZElHoAWXIACjD6BABKegBeQ1NzK3IAOgA1bE9vYSyC2ITk1IzSHL08wpKI8uiauq9nJvyWznsOYjSASXVeDCz8qfo2eITOMqjyYvp1GAB3ejcPEeFfSRVA5RkQ7T19ZYnN/qjSPYPj0+GGmEv/HJFLdVBoHgZnqsAL40GFxXjqYRQdR1ei4GB1f6JD4/eo+PzXYHBMEAFUMIDEVwCRLupMMhwAFjBYPQSfQKWRWKs2hB+My+JYRDN6LU8Y02YsYPgVhsOOtXhxeCZ6FlJfh6LxcPsAK5KGUKtarA0YBlQCDHL70ACCUAA5tr8DBEQA5XUKACiCDAMC4oPUWRRjogJlVwml4yNnDhCThcJoCKRKIU9GwwFwGCg2Bk2MpgJuxK0ZK5ssSKVT6cz2eFovOMCyErDEeoUKAA===) some movs, however, that's not the case here.